inkandswitch/capstone
# Basic sidecar app

Adds a basic version of the sidecar app.

Also fixes the many `Unknown port id:` errors that appear in the background-page log.

The app's entry files live in `src/apps/sidecar/`, I didn't move the capstone entry files in this PR, but they should probably live in `src/apps/capstone/`.

The app builds to `dist/sidecar`, it can be loaded as an unpacked extension just like the capstone app.

Testing:

- On a separate device from _capstone_, run the build with `yarn start`.
- Load `dist/sidecar` as an unpacked extension.
- Run _sidecar_ (from `chrome:apps` on mac).
- In the _capstone_'s console, run `chrome.storage.local.get('workspaceUrl', console.log)`.
- Copy the logged url that looks like this: `capstone://Workspace/GXs3J3FWpBonBoXh6HKyEP28Uuu1V3DNbV6M3jawPgsk/2vh`.
- Send that url to the _sidecar_ device, paste it into the displayed text input, and press enter.
- The _sidecar_ app displays a dashed border when it's ready.
- Drop txt files or paste text into the _sidecar_ app.
- See that those files appear in the _capstone_ archive (three finger swipe down).

Notes:

I had to change the `ChangeFn<T>` input and return types back to `EditDoc<T>`. Without this, all properties inside a change block are readonly (except for props defined in `Model`, which are mutable. But in the case of creating a new Text card, I don't have a Model).
